File: Serene/Serene.Script/Imports/FormContexts/Northwind.CategoryForm.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class CategoryForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Category";
    
        public CategoryForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor CategoryName { [InlineCode("{this}.w('CategoryName', Serenity.StringEditor)")] get; private set; }
        public StringEditor Description { [InlineCode("{this}.w('Description', Serenity.StringEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Northwind.CategoryForm.ts
namespace Serene.Northwind {
    export class CategoryForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Category';

    }

    export interface CategoryForm extends Serenity.PrefixedContext {
        CategoryName(): Serenity.StringEditor;
        Description(): Serenity.StringEditor;
    }

    [['CategoryName', Serenity.StringEditor], ['Description', Serenity.StringEditor]].forEach(x => CategoryForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

